refactor: streamline cluster analysis process

sjungling opened the pull request:
### Overview:
Consolidate python methods and files to form a CLI to help offer easier discoverability of how to analyze either local logs or remote logs from Artifactory. Also switches to a smaller model for smaller, less memory/disk intensive, and faster results; results between two felt comparable. Rewrote direction for installation and running to support one of four different ways to configure: DevContainer, plain Docker, system Python, or with `uv`.

Optimized various parts in part through the intermediate pickle files that were being written and read between steps.

sjungling committed:
refactor: streamline cluster analysis process

consolidate python methods and treat the script as a CLI to help offer easier discoverability for
how to either analyze local logs or remote logs from artifactory

sjungling committed:
refactor: remove :pickle: file storage between steps

sjungling committed:
refactor: additional clean-up and polish

move from large to small model for embedding, fix issues with cleaning volume mounts

sjungling committed:
docs: update documentation

justine-gehring commented on scripts/download_model.py:
```diff
@@ -1,5 +1,13 @@
-tokenizer = AutoTokenizer.from_pretrained("BAAI/bge-large-en-v1.5")
-model = AutoModel.from_pretrained("BAAI/bge-large-en-v1.5")
+tokenizer = AutoTokenizer.from_pretrained("BAAI/bge-small-en-v1.5")
+model = AutoModel.from_pretrained("BAAI/bge-small-en-v1.5")
```
no downgrade in performance? My gut feeling says there shouldn't be anything very noticable.

justine-gehring replied:
ah just read "results between two felt comparable". Nice

sjungling replied:
yep, it was close enough from my testing that it was worth it for the time/mem/disk savings.

natedanner commented on README.md:
```diff
@@ -1,133 +1,91 @@
-* A Devcontainer compatible client (GitHub Codespaces, GitPod, DevPod, Docker Desktop, Visual Studio Code, etc)
-* Optional:
-  * [Git](https://git-scm.com/downloads)
+Before you begin, you will need to complete one the setup methods in [LOCAL_INSTALL.md](LOCAL_INSTALL.md). This will ensure that you have all the necessary dependencies installed.
```
```suggestion
Before you begin, you will need to complete one of the setup methods in [LOCAL_INSTALL.md](LOCAL_INSTALL.md). This will ensure that you have all the necessary dependencies installed.
```

natedanner commented on README.md:
```diff
@@ -1,133 +1,91 @@
-  * [Git](https://git-scm.com/downloads)
+Before you begin, you will need to complete one the setup methods in [LOCAL_INSTALL.md](LOCAL_INSTALL.md). This will ensure that you have all the necessary dependencies installed.
+1. [Using System Python with `venv`](LOCAL_INSTALL.md#using-system-python-with-venv)
+2. [Using uv](LOCAL_INSTALL.md#using-uv-fast-python-package-installer) (Fast Python Package Installer)
```
Do you think we should make this #1 and our recommendation?

sjungling replied:
as much as I'd like to see it be the recommended path I think there's something to be said for keeping vanilla

shanman190 replied:
I'm sure that Kyle would love to see Devcontainer be the first one. 😆

sjungling replied:
@zieka would probably be pushing for 2️⃣ these days!

sjungling committed:
refactor: add `--refresh` flag to skip download and log extraction

supports iterating on existing set of logs

sjungling committed:
refactor: cleanup docker file

move model download up in the process to avoid possible cache layer invalidation

sjungling committed:
fix: normalize urls to avoid redundant `/`

sjungling committed:
refactor: organize cli around two subcommands: `download` and `analyze`

provides seam between two common actions of either downloading logs from Artifactory or running
analysis. Analysis subcommand also provides means for either sourcing logs from a local directory
instead of artifactory. In addition support for iterating  on existing logs that have been process
and are being manually  marked as  resolved

sjungling committed:
fix: improved error handling around embeddings

also utilize click's echo and prompting

sjungling committed:
docs: clean-up readme

sjungling committed:
fix: `urljoin` will drop the context path from the URL when present

probably not what we'd want

shanman190 reviewed (approved):
Verified that the Devcontainer workflow for `download` and `analyze` works correctly.
